### ISSUE TITLE
Iteration Plan for May 2019

### ISSUE THREAD

**Issue by patniko** [microsoft/appcenter#361]
# May 2019 Plan 

Last month, we hit most of our goals for the period. We delivered updates to our **Build** service to add support for the latest versions of Xcode and macOS plus updates to Mono and Xamarin support. We enabled search for handled errors in **Diagnostics**, changed the way **Distribution** updates work on a device in guided access mode, and enhanced **Test** to support NUnit 3. We also delivered the REST API and UI for the new Push to User capabilities in the **Push** service.

On the MBaaS side, we completed the UI for the public preview for our new **Auth** and **Data** services; you should be able to start using these services in your apps in early May. Our work on these services moved rather quickly considering we started work at the beginning of the year. Our intent is to ship early, so you can try them out yourselves, and continue to iterate on new features as we use customer feedback to tune our plans.

Our **Build** service gets some special attention this month as well, we expect to make progress on the clone branch configurations feature and start investigating adding support for git and GitLab. We’re also finally ready to roll out the first version of our HockeyApp migration experience so our existing HockeyApp customers can begin transitioning fully on to Visual Studio App Center.

### HockeyApp Migration Experience
- [x] Opening private preview for interested users [#230]

### Auth 
- [x] Ship early public preview [#388]

### Build
- [x] Clone branch configurations [#22]
- [x] Investigate gitlab.com and/or general git support [#37]

### Data
- [x] Ship early public preview

### Diagnostics
- [x] Filter crash and error groups by build number [#133]

### Distribution
- [x] CodePush support for Electron apps 

### General
- [ ] AAD groups support in Organizations
- [ ] Add a Member role to Organizations

**Comment by lassana** [comment]
Could you consider #30 (Xamarin.iOS entension signing)?

**Comment by SVCAppCenterGit** [comment]
Are you considering supporting TestNG ?

**Comment by jamelait** [comment]
Is there currently a way to send analytics data and crash reports to AppCenter from an Electron app?

**Comment by jwhiteDev** [comment]
@SVCAppCenterGit no plans at this time, though we are certainly monitoring interest on various test frameworks. Always open to feedback/ideas.

**Comment by patniko** [comment]
@jamelait definitely on our mind, but still have a lot to do with our current platforms. I did mess around with it a bit [here](https://github.com/patniko/appcenter-sdk-node/blob/master/analytics.ts).

**Comment by brimmy11** [comment]
Where can I find further information on how I can consume Push to User? I understand it is not part of the SDK yet, but cannot find any information on it at your REST API explorer either (https://openapi.appcenter.ms/).

**Comment by patniko** [comment]
@brimmy11 have you looked at this documentation? 

https://docs.microsoft.com/en-us/appcenter/push/push-to-user

**Comment by brimmy11** [comment]
@patniko that wasn't what I was after, I was looking for a way to target a user via the REST API. I only just noticed you can toggle between "Example Value" and "Model" on the OpenAPI Explorer. Currently we target "devices" (installation id's), now I can see there is a "user_ids" type. I think I should be ok now :)

**Comment by patniko** [comment]
Excellent! Will close this for now, but please do let us know if you need anything else.

**Comment by ttirrell** [comment]
I currently am using Auth through Azure, but would like to transition to App Center (which I know still uses Azure).  Are there plans to support UWP and MacOS?

What about Data for UWP and MacOS?

Thanks for the great work!!

**Comment by ttirrell** [comment]
Also, Is there a place for current videos / training?

**Comment by amchew** [comment]
Hi @ttirrell, 

Thanks for the kind words and support! We have plans to support UWP in our roadmap but there’s no exact timeline yet; we currently support iOS, Android and Xamarin, and have a work item that will be worked on this summer to support React Native. MacOS is not on our roadmap yet but our work is also based on our developer community that we serve, so if we hear feedback, we will continually reprioritize our efforts. 

Re videos and training, not yet. However that is a good point and we’ll work on it. 

You mentioned that you’re using Auth through Azure. Would you mind sharing what service you’re using, and your use case?

**Comment by danieljgmaclean** [comment]
Any chance you could support "APK per ABI through Release/Distribute" for Xamarin Android any time soon? #99

**Comment by chris-cubhub** [comment]
We're loving AppCenter so far and we are fully on board with it for all features. However, the upcoming restrictions for distribution will drastically reduce it's value for us since we'll have to go back to manually building and deploying droid. Are there plans in the next month or two to get support for .aab support so we can keep our automated distribution pipelines in place?

I've already voted here: https://github.com/microsoft/appcenter/issues/194 but was hoping a bump on this board might also assist in planning for both of us.

From google: "By August 1, 2019, all apps that use native code must provide a 64-bit version in addition to the 32-bit version in order to publish an update. This past January, we reiterated that this is required in order to make way for innovation and in anticipation of future Android devices that only support 64-bit code."
